src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACTerminalUnitVariableRefrigerantFlow.cpp

```diff
@@ -264,121 +249,138 @@ namespace energyplus {
-          _supplementalHeatingCoil->setString(Coil_Heating_ElectricFields::AirInletNodeName, fanOutletNodeName);
-          _supplementalHeatingCoil->setString(Coil_Heating_ElectricFields::AirOutletNodeName, outletNodeName);
-        } else if (_supplementalHeatingCoil->iddObject().type() == IddObjectType::Coil_Heating_Water) {
-          _supplementalHeatingCoil->setString(Coil_Heating_WaterFields::AirInletNodeName, fanOutletNodeName);
-          _supplementalHeatingCoil->setString(Coil_Heating_WaterFields::AirOutletNodeName, outletNodeName);
-        } else if (_supplementalHeatingCoil->iddObject().type() == IddObjectType::Coil_Heating_Steam) {
-          // Not yet supported in OS
-          OS_ASSERT(false);
-        } else {
-          LOG(Error, "Unsupported supplemental heating Coil type for " << modelObject.briefDescription());
-          OS_ASSERT(false);
-        }
+        auto getCoilInletOutletIndexes = [&modelObject](const IdfObject& idf_coil) {
+          if (idf_coil.iddObject().type() == IddObjectType::Coil_Heating_Fuel) {
+            return std::pair<unsigned, unsigned>{Coil_Heating_FuelFields::AirInletNodeName, Coil_Heating_FuelFields::AirOutletNodeName};
+          } else if (idf_coil.iddObject().type() == IddObjectType::Coil_Heating_Electric) {
+            return std::pair<unsigned, unsigned>{Coil_Heating_ElectricFields::AirInletNodeName, Coil_Heating_ElectricFields::AirOutletNodeName};
+          } else if (idf_coil.iddObject().type() == IddObjectType::Coil_Heating_Water) {
+            return std::pair<unsigned, unsigned>{Coil_Heating_WaterFields::AirInletNodeName, Coil_Heating_WaterFields::AirOutletNodeName};
+          } else if (idf_coil.iddObject().type() == IddObjectType::Coil_Heating_Steam) {
+            // Not yet supported in OS
+            OS_ASSERT(false);
+          } else {
+            LOG(Error, "Unsupported supplemental heating Coil type for " << modelObject.briefDescription());
+            OS_ASSERT(false);
+          }
+          return std::pair<unsigned, unsigned>{0, 0};
+        };
+
+        auto [inletIndex, outletIndex] = getCoilInletOutletIndexes(_supplementalHeatingCoil.get());
+
+        compsInOrder.emplace_back(
+          Component(_supplementalHeatingCoil.get(), modelObject.name().get() + " Supplemental Heating Coil Outlet Node", inletIndex, outletIndex));
+      }
+    }
+
+    // Now we write the nodes in the order we determine
+    // Last Node is the VRT TU outlet Node
+    compsInOrder.back().outletNodeName = outletNodeName;
+
+    size_t curIndex = 0;
+    for (auto& comp : compsInOrder) {
+      comp.idfObject.setString(comp.inletIndex, inletNodeName);
+      comp.idfObject.setString(comp.outletIndex, comp.outletNodeName);
+
+      if (curIndex == fanIndex) {
+        fixSPMsForUnitarySystem(modelObject, inletNodeName, comp.outletNodeName);
       }
+
+      inletNodeName = comp.outletNodeName;
+      ++curIndex;
```

Comment:
Write the nodes.

There's a special case... for the fan, we also call `fixSPMsForUnitarySystem` so that the SPM:MixerAir will correctly reference the VRT TU Fan Inlet/Outlet Node names